src/sentry/remote_config/endpoints.py

```diff
@@ -0,0 +1,89 @@
+from rest_framework import serializers
+from rest_framework.request import Request
+from rest_framework.response import Response
+from rest_framework.serializers import Serializer
+
+from sentry import features
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases.project import ProjectEndpoint, ProjectEventPermission
+from sentry.api.exceptions import ResourceDoesNotExist
+from sentry.models.project import Project
+from sentry.models.projectkey import ProjectKey
+from sentry.remote_config.storage import make_storage
+
+
+class ConfigurationValidator(Serializer):
+    id = serializers.UUIDField(read_only=True)
+    sample_rate = serializers.FloatField(max_value=1.0, min_value=0, required=True)
+    traces_sample_rate = serializers.FloatField(max_value=1.0, min_value=0, required=True)
+    user_config = serializers.JSONField(required=True, allow_null=True)
+
+
+class ConfigurationContainerValidator(Serializer):
+    data = ConfigurationValidator()  # type: ignore[assignment]
+
+
+@region_silo_endpoint
+class ProjectConfigurationEndpoint(ProjectEndpoint):
```

Comment:
If a configuration belongs to a key the way the url represents it, the endpoint is better be called `ProjectKeyConfigurationEndpoint`